Document Submariner's role in the ecosystem (#1151)
submariner-io/submariner-website: src/content/community/role-ecosystem/_index.en.md

+++
title = "Role in the Ecosystem"

> date = 2020-02-19T20:24:34+01:00

Reviewer: nit: date

“namespace seamness” principal: Kubernetes namespaces behave consistently and seamlessly across interconnected clusters.

**Use Cases**: It is well-suited for scenarios where you need to create a unified permanent network across geographically distributed
clusters, ensuring seamless pod-to-pod communication and service discovery.

**Connectivity Domain**: Submariner primarily focuses on interconnecting Kubernetes clusters. However, Submariner also provides an
experimental feature that allows access to external applications or endpoints that exist outside of the cluster, in non-Kubernetes
environments. It's important to note that while this [experimental feature](../../getting-started/quickstart/external/) exists, it might not
be as mature or stable as Submariner's core functionalities. Users interested in leveraging Submariner for accessing external applications
should consider testing and evaluating this feature in their specific use cases.

**Integration**: Submariner integrates with various networking solutions and can work alongside existing CNI
(Container Network Interface) plug-ins like Calico, Flannel, etc., ensuring compatibility with different Kubernetes environments.

> **Security**: Provides secure communication between clusters using IPsec tunnels by default, which encrypt traffic between pods and services
> across clusters.

Reviewer: The sentence is probably giving an impression that pod to pod communication is encrypted. Do you think we can rephrase it as follows?

```suggestion
**Security**: Provides secure communication between clusters using IPsec tunnels by default, which encrypt traffic between the clusters.
```

years. There are several community projects and emerging technologies addressing various challenges of connecting, securing, and managing
communication between Kubernetes clusters across different environments. Some notable examples include
[Submariner](https://submariner.io/), [Skupper](https://skupper.io/), [Istio](https://istio.io/),
[Calico](https://www.tigera.io/project-calico/community/), and [Cilium](https://cilium.io/). These solutions play a critical role in
enabling scalable and secure multi-cluster deployments.

This page highlights Submariner’s role within the ecosystem and touches on how it compares to some of these other projects.

## Submariner

**Networking Focus**: Submariner is primarily focused on providing network connectivity at layer 3. It establishes tunnels between clusters
to facilitate direct communication between pods and services. Submariner operates at layer 3 of the OSI model, which means it can support
communication for any type of application data or protocol. However, setting up Submariner does require some administrative overhead,
particularly in configuring firewall rules in the underlying infrastructure.

> **Service Discovery**: Submariner provides an implementation of the Multicluster Services API (MCS API), an initiative within the Kubernetes

Reviewer 2: ```suggestion
**Service Discovery**: Submariner provides an implementation of the Multi-cluster Services API (MCS API), an initiative within the Kubernetes
```

> ecosystem aimed at standardizing the management of services across multiple Kubernetes clusters, and follows its model and core
> “namespace seamness” principal: Kubernetes namespaces behave consistently and seamlessly across interconnected clusters.

Reviewer 2: ```suggestion
ecosystem aimed at standardizing the management of services across multiple Kubernetes clusters, and follows its core principal of
“namespace sameness” whereby Kubernetes namespaces behave consistently and seamlessly across interconnected clusters.
```

experimental feature that allows access to external applications or endpoints that exist outside of the cluster, in non-Kubernetes
environments. It's important to note that while this [experimental feature](../../getting-started/quickstart/external/) exists, it might not
be as mature or stable as Submariner's core functionalities. Users interested in leveraging Submariner for accessing external applications
should consider testing and evaluating this feature in their specific use cases.

**Integration**: Submariner integrates with various networking solutions and can work alongside existing CNI
(Container Network Interface) plug-ins like Calico, Flannel, etc., ensuring compatibility with different Kubernetes environments.

**Security**: Provides secure communication between clusters using IPsec tunnels by default, which encrypt traffic between clusters.

## Comparison Summary

- **Submariner** focuses on establishing a unified network between Kubernetes clusters, ensuring secure pod-to-pod communication and service
discovery.

- **Skupper** leverages messaging functionalities to facilitate flexible communication across end-points. While Skupper provides support for

> linking namespaces across different Kubernetes clusters, it can also be used to support non-Kubernetes environments, including for example

Reviewer 2: I don't think we need "for example" in this context.

```suggestion
linking namespaces across different Kubernetes clusters, it can also be used to support non-Kubernetes environments, including
```
